enclave_build/src/storage.rs

```diff
@@ -398,10 +398,38 @@ impl OciStorage {
 
         docker_prefix.to_owned() + reference
     }
+
+    /// Fetches the image metadata from storage as an ImageDetails struct.
+    ///
+    /// If the data is not correctly stored or a file is missing, it returns an error.
+    ///
+    /// If the image is not stored, it does not attempt to pull the image from remote.
+    pub fn fetch_image_details(&mut self, image_name: &str) -> Result<ImageDetails> {
+        if self.config.is_none() {
+            self.config = Some(self.check_stored_image(image_name)?);
+        }
+        let image_config = self.config.as_ref().ok_or(EnclaveBuildError::ConfigError)?;
+
+        // Add algorithm prefix to the hash
+        let image_hash = format!(
+            "sha256:{:x}",
+            sha2::Sha256::digest(
+                &serde_json::to_vec(&self.config).map_err(EnclaveBuildError::SerdeError)?
+            )
+        );
```

Comment:
I'm not sure if I suggested at some point to not have this as a separate function, probably I did that, sorry, but looking again how many times this is used, I think it would be better though to define one, so that in case we change the algorithm we won't need to update it in multiple places.